src/charm.py

```diff
@@ -146,8 +146,9 @@ def _deploy_k8s_resources(self) -> None:
             self.unit.status = MaintenanceStatus("Creating K8S resources")
             self.k8s_resource_handler.apply()
             self.crd_resource_handler.apply()
-        except ApiError:
-            raise ErrorWithStatus("K8S resources creation failed", BlockedStatus)
+        except ApiError as e:
+            self.logger.exception(f"Failed to create K8S resources, with error: {e}")
+            raise e
```

Comment:
```suggestion
        except ApiError:
            self.logger.error(f"Failed to create K8S resources")
            raise
```

Reply:
We should include {e} in the log message. It will help in debugging.

Reply:
see https://github.com/canonical/training-operator/pull/83#discussion_r1137107037